test/e2e/addon/podidentityverify.go

```diff
@@ -45,7 +50,49 @@ func (v VerifyPodIdentityAddon) Run(ctx context.Context) error {
 		return err
 	}
 
-	// TODO: Deploy a pod with service account to verify the pod identity token section is populated in pod manifest files
+	// Deploy a pod with service account to verify the pod identity token section is populated in pod yaml
+	node, err := kubernetes.WaitForNode(ctx, v.K8S, v.NodeIP, v.Logger)
+	if err != nil {
+		return err
+	}
+
+	nodeName := node.Name
+	podName := kubernetes.GetNginxPodName(nodeName)
+	v.Logger.Info("Creating a test pod on the hybrid node for pod identity add-on")
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "nginx",
+					Image: fmt.Sprintf("%s.dkr.ecr.%s.amazonaws.com/ecr-public/nginx/nginx:latest", constants.EcrAccounId, "us-west-2"),
```

Comment:
can we use the same region as the cluster? the more cross region traffic we avoid the better

Reply:
if not, can use public ecr?